src/components/CreatePost/Location.tsx

```diff
@@ -1,26 +1,30 @@
 import { LocationOnOutlined } from '@mui/icons-material';
 import { Box, IconButton, OutlinedInput } from '@mui/material';
-import { ChangeEvent, useState } from 'react';
+import { useEffect } from 'react';
+import { ControllerRenderProps } from 'react-hook-form';
+
+import { CreatePost } from '@/types/CreatePost';
 
 interface LocationProps {
   readonly?: boolean;
+  name?: ControllerRenderProps<CreatePost, 'country.name'>;
 }
 
-const Location = ({ readonly }: LocationProps) => {
-  const [value, setValue] = useState('');
-  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setValue(e.target.value);
-  };
+const Location = ({ readonly, name }: LocationProps) => {
+  useEffect(() => {
+    if (name) {
+      readonly ? name.onChange('대한민국') : name.onChange('');
+    }
+  }, [readonly]);
```

Comment:
p1;

```suggestion
  }, [name, readonly]);
```

useEffect에서 name props 의존성이 있으니 추가해주세요 👍

Reply:
ask;

```tsx
<Stack sx={marginBottom} mb={1}>
```

`@/components/CreatePost/Location.tsx`에서 68번째 줄에서 `marginBottom 변수`로 분리한 이유가 있나요? 혹은 `mb={1}`을 사용하지 않은 이유가 궁금해요.

Reply:
name을 추가하면 name 값이 바뀌면 저부분이 호출되기때문에 name.onChange('')가 호출됩니다.
따라서 값을 입력해도 ''으로 바뀌어서 일단 빼두었어요

Reply:
marginBottom같은 단일스타일이 다른곳에서도 동일하게 사용되는데 나중에 한번에 바꾸겠습니다.
(원래는 다른 스타일도 추가될것같아서 따로 빼두었어요 디자인이 완료된 단계라 이제 빼도 될것같습니다.)